src/ctl.c

```diff
@@ -3083,3 +3085,76 @@ experimental_hooks_remove_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
 label_return:
 	return ret;
 }
+
+static int
+experimental_frag_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret = 0;
+
+	/* Requires both read and write */
+	if (oldp == NULL || oldlenp == NULL || newp == NULL || newlen == 0) {
+		ret = EINVAL;
+		goto label_return;
+	}
+
+	/*
+	 * Make sure newlen is an exact multiple of sizeof(const void *),
+	 * and if so, reduce it to the multiple i.e. count of input pointers.
+	 * Rely on compiler to convert multiplication / division to shifts.
+	 */
+	if (newlen != newlen / sizeof(const void *) * sizeof(const void *)) {
+		ret = EINVAL;
+		goto label_return;
+	}
+	newlen /= sizeof(const void *);
```

Comment:
nit: Similarly, keep the original input args, feel free to define a new one with explicit naming.